presidio-image-redactor/presidio_image_redactor/bbox.py

```diff
@@ -0,0 +1,135 @@
+from presidio_image_redactor.entities import ImageRecognizerResult
+from typing import List, Tuple, Dict, Union
+
+
+class BboxProcessor:
+    """Common module for general bounding box operators."""
+
+    def get_bboxes_from_ocr_results(
+        self,
+        ocr_results: Dict[str, List[Union[int, str]]],
+    ) -> List[Dict[str, Union[int, float, str]]]:
+        """Get bounding boxes on padded image for all detected words from ocr_results.
+
+        :param ocr_results: Raw results from OCR.
+        :return: Bounding box information per word.
+        """
+        bboxes = []
+        item_count = 0
+        for i in range(len(ocr_results["text"])):
+            detected_text = ocr_results["text"][i]
+            if detected_text:
+                bbox = {
+                    "left": ocr_results["left"][i],
+                    "top": ocr_results["top"][i],
+                    "width": ocr_results["width"][i],
+                    "height": ocr_results["height"][i],
+                    "conf": float(ocr_results["conf"][i]),
+                    "label": detected_text,
```

Comment:
I would consider consolidating those attributes in a Bbox class

Reply:
Besides reducing some lines of code (lines 23-28 and 50-55), is there a functional reason to create an additional class? I feel that this would be extra work that would also need extra tests.

Reply:
the spread operator / ocr_result to bbox implemented once

Reply:
Could you provide a code example? I don't think I'm visualizing the correct implementation in terms of how the bbox class would be structured and implemented.

Reply:
Assuming Bbox is defined as dict or pydantic or dataclass:
```python
class Bbox(dict):
    left: int
    top: int
    width: int
    height: int
    text: str
    conf: float
    label: str
    entity_type: str

```

We can transform `get_bboxes_from_analyzer_results` to a one-liner:
```python
    def get_bboxes_from_analyzer_results(
        self,
        analyzer_results: List[ImageRecognizerResult],
    ) -> List[Dict[str, Union[str, float, int]]]:
        """Organize bounding box info from analyzer results.

        :param analyzer_results: Results from using ImageAnalyzerEngine.

        :return: Bounding box info organized.
        """
        return [Bbox(**result.to_dict()) for result in analyzer_results]

```

And Bbox an be accessed with a more straightforward access `bbox.left `instead of `bbox["left"]`

Reply:
Got it! Thank you for explicitly writing this out. This helps a lot. I'll make the changes.

Reply:
@SharonHart - I've spent this morning making the changes in my local branch and have discovered that introducing the `Bbox` class introduces more complications than I believe are necessary.

## Complications
1. The `get_bboxes_from_analyzer_results` method cannot be reduced down to a single line because this method is also used to filter out irrelevant properties and only carry over relevant information into the bounding boxes. The method would need to look like the following if we introduce the `Bbox` class.

```python
    def get_bboxes_from_analyzer_results(
        self,
        analyzer_results: List[ImageRecognizerResult],
    ) -> List[Bbox]:
        """Organize bounding box info from analyzer results.

        :param analyzer_results: Results from using ImageAnalyzerEngine.

        :return: Bounding box info organized.
        """
        bboxes = []
        exclude = ["start", "end", "analysis_explanation", "recognition_metadata"]
        for result in analyzer_results:
            values = result.to_dict()

            # Remove irrelevant fields
            for key in exclude:
                values.pop(key, None)

            # Add bbox with relevant fields
            bboxes.append(Bbox(**values))

        return bboxes
```

2. Users provide the ground truth as a dictionary either defined in their notebook or loaded from a json file during evaluation. Loading ground truth from JSON files is convenient and works easily with dictionaries. If we use the `Bbox` class for OCR results and analyzer results, but then need to compare those against the ground truth (like in `BboxProcessor.match_with_source`), there is a mix of types. Keeping everything as dictionaries is easier and more straightforward in my opinion. Forcing users to format their ground truth to fit the `Bbox` class would be an unnecessary extra step and prevent easy use with JSON files.
4. There would be no methods or any operations needed for the `Bbox` class itself, which I feel supports the case to not make a class for it in the first place.

## Suggestion
Given the above, I suggest we keep the code as-is with the latest commit. If that is alright with you, please re-approve and I will resolve the related comments.